modules/ocf_backups/manifests/offsite.pp

```diff
@@ -19,5 +28,25 @@
     '/opt/share/backups/boto.cfg':
       source => 'puppet:///private/boto.cfg',
       mode   => '0600';
+
+    # Box.com credentials and API id/secret
+    '/opt/share/backups/box-creds.json':
+      source => 'puppet:///private/box-creds.json',
+      mode   => '0600';
+
+    # This changes each time a backup is made, so it is not stored on the
+    # puppetmaster and instead is just a local file on the backups server.
+    '/opt/share/backups/box-refresh-token':
+      mode => '0600';
```

Comment:
I would like it if Puppet didn't error on the first run too, but that would also mean backups wouldn't work once they run on that week (or the uploaded folder wouldn't be shared correctly at least, the encryption and upload would still work). Unless a human had copied the refresh token from the old server (currently hal) and added it on the new server, then it doesn't have access to the Box.com API.

The issue is that as part of OAuth for Box, whenever a new access token is claimed using the old refresh token, a new refresh token is sent back too, and this can happen indefinitely as long as each refresh token is used within 60 days of being issued. However, that means that the new refresh token has to be stored somewhere until the next run. Storing it in the puppetmaster's private store would be best, but I don't see a way to do that without having to ssh in or something.